--- /Users/lion/Downloads/source/js/v8/src/arm/deoptimizer-arm.cc	2014-07-10 13:47:16.000000000 +0800
+++ src/arm/deoptimizer-arm.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "src/codegen.h"
 #include "src/deoptimizer.h"
 #include "src/full-codegen.h"
@@ -355,3 +357,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm/lithium-arm.cc	2014-07-21 03:36:08.000000000 +0800
+++ src/arm/lithium-arm.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "src/arm/lithium-arm.h"
 #include "src/arm/lithium-codegen-arm.h"
 #include "src/hydrogen-osr.h"
@@ -2596,3 +2598,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm/lithium-codegen-arm.cc	2014-07-21 03:36:08.000000000 +0800
+++ src/arm/lithium-codegen-arm.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "src/arm/lithium-codegen-arm.h"
 #include "src/arm/lithium-gap-resolver-arm.h"
 #include "src/code-stubs.h"
@@ -5886,3 +5888,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm/lithium-gap-resolver-arm.cc	2014-06-22 19:23:17.000000000 +0800
+++ src/arm/lithium-gap-resolver-arm.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "src/arm/lithium-codegen-arm.h"
 #include "src/arm/lithium-gap-resolver-arm.h"
 
@@ -300,3 +302,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/deoptimizer-arm64.cc	2014-07-21 03:36:08.000000000 +0800
+++ src/arm64/deoptimizer-arm64.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "src/codegen.h"
 #include "src/deoptimizer.h"
 #include "src/full-codegen.h"
@@ -357,3 +359,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/instrument-arm64.cc	2014-06-09 04:42:32.000000000 +0800
+++ src/arm64/instrument-arm64.cc	2014-07-21 03:44:10.000000000 +0800
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "src/v8.h"
+
+#if V8_TARGET_ARCH_ARM64
+
 #include "src/arm64/instrument-arm64.h"
 
 namespace v8 {
@@ -593,3 +597,5 @@
 
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/lithium-arm64.cc	2014-07-21 03:36:08.000000000 +0800
+++ src/arm64/lithium-arm64.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "src/arm64/lithium-arm64.h"
 #include "src/arm64/lithium-codegen-arm64.h"
 #include "src/hydrogen-osr.h"
@@ -2726,3 +2728,5 @@
 
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/lithium-codegen-arm64.cc	2014-07-21 03:36:08.000000000 +0800
+++ src/arm64/lithium-codegen-arm64.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "src/arm64/lithium-codegen-arm64.h"
 #include "src/arm64/lithium-gap-resolver-arm64.h"
 #include "src/code-stubs.h"
@@ -6049,3 +6051,5 @@
 
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/lithium-gap-resolver-arm64.cc	2014-06-29 00:42:36.000000000 +0800
+++ src/arm64/lithium-gap-resolver-arm64.cc	2014-07-21 03:44:10.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "src/v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "src/arm64/delayed-masm-arm64-inl.h"
 #include "src/arm64/lithium-codegen-arm64.h"
 #include "src/arm64/lithium-gap-resolver-arm64.h"
@@ -293,3 +295,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-freebsd.cc	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform-freebsd.cc	2014-07-21 03:44:10.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for FreeBSD goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef FreeBSD
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <signal.h>
@@ -305,3 +309,5 @@
 }
 
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-linux.cc	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform-linux.cc	2014-07-21 03:44:10.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for Linux goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef Linux
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <signal.h>
@@ -430,3 +434,5 @@
 }
 
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-macos.cc	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform-macos.cc	2014-07-21 03:44:10.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for MacOS goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef MacOS
+
 #include <dlfcn.h>
 #include <mach/mach_init.h>
 #include <mach-o/dyld.h>
@@ -308,3 +312,5 @@
 }
 
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-openbsd.cc	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform-openbsd.cc	2014-07-21 03:44:10.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for OpenBSD and NetBSD goes here. For the
 // POSIX-compatible parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef OpenBSD
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <signal.h>
@@ -336,3 +340,5 @@
 }
 
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-posix.cc	2014-07-10 13:47:16.000000000 +0800
+++ src/base/platform/platform-posix.cc	2014-07-21 03:44:10.000000000 +0800
@@ -1,3 +1,6 @@
+
+#ifndef WIN32
+
 // Copyright 2012 the V8 project authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
@@ -267,12 +270,12 @@
   return static_cast<size_t>(sysconf(_SC_PAGESIZE));
 }
 
-
+#if 0
 void OS::Sleep(int milliseconds) {
   useconds_t ms = static_cast<useconds_t>(milliseconds);
   usleep(1000 * ms);
 }
-
+#endif
 
 void OS::Abort() {
   if (g_hard_abort) {
@@ -498,7 +501,7 @@
   strncpy(dest, src, n);
 }
 
-
+#if 0
 // ----------------------------------------------------------------------------
 // POSIX thread support.
 //
@@ -733,5 +736,7 @@
   USE(result);
 }
 
-
+#endif
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-qnx.cc	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform-qnx.cc	2014-07-21 03:44:10.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for QNX goes here. For the POSIX-compatible
 // parts the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+ 
+#ifdef Qnx
+ 
 #include <backtrace.h>
 #include <pthread.h>
 #include <semaphore.h>
@@ -372,3 +376,5 @@
 }
 
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-solaris.cc	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform-solaris.cc	2014-07-21 03:44:10.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for Solaris 10 goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef Solaris
+
 #ifdef __sparc
 # error "V8 does not support the SPARC CPU architecture."
 #endif
@@ -277,3 +281,5 @@
 }
 
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform-win32.cc	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform-win32.cc	2014-07-21 03:44:10.000000000 +0800
@@ -1,3 +1,6 @@
+
+#ifdef WIN32
+
 // Copyright 2012 the V8 project authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
@@ -820,11 +823,11 @@
   VirtualProtect(address, size, PAGE_NOACCESS, &oldprotect);
 }
 
-
+#if 0
 void OS::Sleep(int milliseconds) {
   ::Sleep(milliseconds);
 }
-
+#endif
 
 void OS::Abort() {
   if (g_hard_abort) {
@@ -1323,7 +1326,7 @@
   return false;
 }
 
-
+#if 0
 // ----------------------------------------------------------------------------
 // Win32 thread support.
 
@@ -1425,5 +1428,7 @@
 void Thread::YieldCPU() {
   Sleep(0);
 }
-
+#endif
 } }  // namespace v8::base
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/base/platform/platform.h	2014-07-05 18:20:51.000000000 +0800
+++ src/base/platform/platform.h	2014-07-21 03:44:10.000000000 +0800
@@ -77,6 +77,8 @@
 // ----------------------------------------------------------------------------
 // Fast TLS support
 
+#define V8_NO_FAST_TLS
+
 #ifndef V8_NO_FAST_TLS
 
 #if defined(_MSC_VER) && (V8_HOST_ARCH_IA32)
