--- /Users/lion/Downloads/source/js/v8/src/arm/deoptimizer-arm.cc	2014-05-27 00:26:03.000000000 +0800
+++ src/arm/deoptimizer-arm.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "codegen.h"
 #include "deoptimizer.h"
 #include "full-codegen.h"
@@ -355,3 +357,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm/lithium-arm.cc	2014-05-29 01:59:38.000000000 +0800
+++ src/arm/lithium-arm.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "lithium-allocator-inl.h"
 #include "arm/lithium-arm.h"
 #include "arm/lithium-codegen-arm.h"
@@ -2594,3 +2596,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm/lithium-codegen-arm.cc	2014-05-29 01:59:38.000000000 +0800
+++ src/arm/lithium-codegen-arm.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "arm/lithium-codegen-arm.h"
 #include "arm/lithium-gap-resolver-arm.h"
 #include "code-stubs.h"
@@ -5827,3 +5829,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm/lithium-gap-resolver-arm.cc	2014-05-03 21:43:54.000000000 +0800
+++ src/arm/lithium-gap-resolver-arm.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM
+
 #include "arm/lithium-gap-resolver-arm.h"
 #include "arm/lithium-codegen-arm.h"
 
@@ -300,3 +302,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/deoptimizer-arm64.cc	2014-05-27 00:26:01.000000000 +0800
+++ src/arm64/deoptimizer-arm64.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "codegen.h"
 #include "deoptimizer.h"
 #include "full-codegen.h"
@@ -383,3 +385,5 @@
 #undef __
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/instrument-arm64.cc	2014-05-03 21:43:44.000000000 +0800
+++ src/arm64/instrument-arm64.cc	2014-05-29 02:08:35.000000000 +0800
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "v8.h"
+
+#if V8_TARGET_ARCH_ARM64
+
 #include "arm64/instrument-arm64.h"
 
 namespace v8 {
@@ -593,3 +597,5 @@
 
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/lithium-arm64.cc	2014-05-29 01:59:37.000000000 +0800
+++ src/arm64/lithium-arm64.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "lithium-allocator-inl.h"
 #include "arm64/lithium-arm64.h"
 #include "arm64/lithium-codegen-arm64.h"
@@ -2722,3 +2724,5 @@
 
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/lithium-codegen-arm64.cc	2014-05-29 01:59:37.000000000 +0800
+++ src/arm64/lithium-codegen-arm64.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "arm64/lithium-codegen-arm64.h"
 #include "arm64/lithium-gap-resolver-arm64.h"
 #include "code-stubs.h"
@@ -6019,3 +6021,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/arm64/lithium-gap-resolver-arm64.cc	2014-05-03 21:43:44.000000000 +0800
+++ src/arm64/lithium-gap-resolver-arm64.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,8 @@
 
 #include "v8.h"
 
+#if V8_TARGET_ARCH_ARM64
+
 #include "arm64/lithium-gap-resolver-arm64.h"
 #include "arm64/lithium-codegen-arm64.h"
 
@@ -309,3 +311,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-freebsd.cc	2014-05-03 21:43:50.000000000 +0800
+++ src/platform-freebsd.cc	2014-05-29 02:08:35.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for FreeBSD goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef FreeBSD
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <signal.h>
@@ -306,3 +310,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-linux.cc	2014-05-03 21:43:43.000000000 +0800
+++ src/platform-linux.cc	2014-05-29 02:08:35.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for Linux goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef Linux
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <signal.h>
@@ -432,3 +436,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-macos.cc	2014-05-03 21:43:50.000000000 +0800
+++ src/platform-macos.cc	2014-05-29 02:08:35.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for MacOS goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef MacOS
+
 #include <dlfcn.h>
 #include <unistd.h>
 #include <sys/mman.h>
@@ -308,3 +312,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-openbsd.cc	2014-05-03 21:43:43.000000000 +0800
+++ src/platform-openbsd.cc	2014-05-29 02:08:35.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for OpenBSD and NetBSD goes here. For the
 // POSIX-compatible parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef OpenBSD
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <signal.h>
@@ -338,3 +342,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-posix.cc	2014-05-29 01:59:38.000000000 +0800
+++ src/platform-posix.cc	2014-05-29 02:08:35.000000000 +0800
@@ -6,6 +6,10 @@
 // own, but contains the parts which are the same across the POSIX platforms
 // Linux, MacOS, FreeBSD, OpenBSD, NetBSD and QNX.
 
+#include <osconfig.h>
+
+#ifndef Windows
+
 #include <dlfcn.h>
 #include <pthread.h>
 #if defined(__DragonFly__) || defined(__FreeBSD__) || defined(__OpenBSD__)
@@ -241,12 +245,12 @@
   return static_cast<size_t>(sysconf(_SC_PAGESIZE));
 }
 
-
+#if 0
 void OS::Sleep(int milliseconds) {
   useconds_t ms = static_cast<useconds_t>(milliseconds);
   usleep(1000 * ms);
 }
-
+#endif
 
 void OS::Abort() {
   if (FLAG_hard_abort) {
@@ -460,7 +464,7 @@
   strncpy(dest.start(), src, n);
 }
 
-
+#if 0
 // ----------------------------------------------------------------------------
 // POSIX thread support.
 //
@@ -694,6 +698,8 @@
   ASSERT_EQ(0, result);
   USE(result);
 }
-
+#endif
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-qnx.cc	2014-05-03 21:43:43.000000000 +0800
+++ src/platform-qnx.cc	2014-05-29 02:08:35.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for QNX goes here. For the POSIX-compatible
 // parts the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef Qnx
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <signal.h>
@@ -375,3 +379,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-solaris.cc	2014-05-03 21:43:44.000000000 +0800
+++ src/platform-solaris.cc	2014-05-29 02:08:35.000000000 +0800
@@ -5,6 +5,10 @@
 // Platform-specific code for Solaris 10 goes here. For the POSIX-compatible
 // parts, the implementation is in platform-posix.cc.
 
+#include <osconfig.h>
+
+#ifdef Solaris
+
 #ifdef __sparc
 # error "V8 does not support the SPARC CPU architecture."
 #endif
@@ -318,3 +322,5 @@
 }
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform-win32.cc	2014-05-29 01:59:37.000000000 +0800
+++ src/platform-win32.cc	2014-05-29 02:08:35.000000000 +0800
@@ -4,6 +4,10 @@
 
 // Platform-specific code for Win32.
 
+#include <osconfig.h>
+
+#ifdef Windows
+
 // Secure API functions are not available using MinGW with msvcrt.dll
 // on Windows XP. Make sure MINGW_HAS_SECURE_API is not defined to
 // disable definition of secure API functions in standard headers that
@@ -801,11 +805,11 @@
   VirtualProtect(address, size, PAGE_NOACCESS, &oldprotect);
 }
 
-
+#if 0
 void OS::Sleep(int milliseconds) {
   ::Sleep(milliseconds);
 }
-
+#endif
 
 void OS::Abort() {
   if (FLAG_hard_abort) {
@@ -1299,7 +1303,7 @@
   return false;
 }
 
-
+#if 0
 // ----------------------------------------------------------------------------
 // Win32 thread support.
 
@@ -1401,5 +1405,8 @@
 void Thread::YieldCPU() {
   Sleep(0);
 }
+#endif
 
 } }  // namespace v8::internal
+
+#endif
--- /Users/lion/Downloads/source/js/v8/src/platform.h	2014-05-29 01:59:37.000000000 +0800
+++ src/platform.h	2014-05-29 02:08:35.000000000 +0800
@@ -76,6 +76,8 @@
 // ----------------------------------------------------------------------------
 // Fast TLS support
 
+#define V8_NO_FAST_TLS
+
 #ifndef V8_NO_FAST_TLS
 
 #if defined(_MSC_VER) && (V8_HOST_ARCH_IA32)
